tests/integration/llm/test_llm_caching_middleware.py

```diff
@@ -0,0 +1,108 @@
+import time
+import unittest
+
+import dotenv
+
+from council import AnthropicLLM
+from council.llm import LLMFunction, LLMCachingMiddleware, LLMResponse
+from council.llm.llm_response_parser import EchoResponseParser
+from council.utils import OsEnviron
+
+
+class TestLlmCachingMiddleware(unittest.TestCase):
+    def setUp(self) -> None:
+        dotenv.load_dotenv()
+        with OsEnviron("ANTHROPIC_LLM_MODEL", "claude-3-haiku-20240307"):
+            self.llm = AnthropicLLM.from_env()
+
+        self.m1 = "What is the capital of France?"
+        self.m2 = "What is the capital of France???"
+        self.m3 = "What is the capital of France?????"
+
+    def get_llm_func(self, ttl: float, cache_limit_size: int):
+        llm_func: LLMFunction[LLMResponse] = LLMFunction(
+            self.llm, EchoResponseParser.from_response, system_message="You're a helpful assistant"
+        )
+
+        llm_func.add_middleware(LLMCachingMiddleware(ttl=ttl, cache_limit_size=cache_limit_size))
+        return llm_func
+
+    @staticmethod
+    def execute_llm_func(llm_func: LLMFunction, message: str, to_print: str, **kwargs) -> LLMResponse:
+        response = llm_func.execute(message, **kwargs)
+        print(f"\n{to_print}")
+        print(f"\tResponse duration: {response.duration:.3f}s")
+
+        return response
+
+    @staticmethod
+    def assert_if_cached(response):
+        assert all(not consumption.unit.startswith("cached_") for consumption in response.result.consumptions)
+
+    @staticmethod
+    def assert_if_not_cached(response):
```

Comment:
I guess I am a bit confused by the name. I would have expect `assert_cached`

Reply:
and the opposite for the previous one

Reply:
True, renamed to `raise_if_cached` / `raise_if_not_cached`